packages/cozy-client/src/hooks/useClientErrors.spec.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import useClientErrors from './useClientErrors'
+import { renderHook, act } from '@testing-library/react-hooks'
+import CozyProvider from '../Provider'
+import { shallow } from 'enzyme'
+import { FetchError } from 'cozy-stack-client'
+
+function createCozyClient() {
+  return {
+    on: jest.fn(),
+    removeListener: jest.fn()
+  }
+}
+
+function createWrapper(client = createCozyClient()) {
+  function Wrapper({ children }) {
+    return <CozyProvider client={client}>{children}</CozyProvider>
+  }
+  return Wrapper
+}
+
+function renderWrappedHook(params, client) {
+  const wrapper = createWrapper(client)
+  return renderHook(() => useClientErrors(), { wrapper })
+}
+
+function wrappedShallow(tree, client) {
+  return shallow(tree, { wrappingComponent: createWrapper(client) })
+}
+
+describe('useClientErrors', () => {
+  it('registers an `error` handler in client', done => {
+    const client = createCozyClient()
+    client.on.mockImplementation(name => name === 'error' && done())
+    renderWrappedHook(undefined, client)
+  })
+
+  describe('renderErrors', () => {
+    it('returns a function', () => {
+      const { result } = renderWrappedHook()
+      const { ClientErrors } = result.current
+      expect(ClientErrors).toBeInstanceOf(Function)
+    })
+
+    it('displays nothing by default', () => {
+      const { result } = renderWrappedHook()
+      const { ClientErrors } = result.current
+      const node = wrappedShallow(<ClientErrors />)
+      expect(node).toHaveLength(0)
+    })
+
+    describe('for quota errors', () => {
+      it('displays a modal', async () => {
+        const client = createCozyClient()
+        const { result, waitForNextUpdate } = renderWrappedHook(
+          undefined,
+          client
+        )
+        const nextUpdate = waitForNextUpdate()
+
+        const response = new Response(null, {
+          status: 413,
+          statusText: 'Quota exceeded'
+        })
+        const error = new FetchError(
+          response,
+          `${response.status} ${response.statusText}`
+        )
+        act(() => {
+          const handler = client.on.mock.calls[0][1]
+          handler(error)
```

Comment:
I am not sure actually that we should have a fake cozy-client here. With a real client, we could do client.emit('error', error) instead of using `.mock.calls[0][1]`.

Reply:
that would test the EventEmitter of Cozy Client. Here I only look for input and outputs of the hook.

Reply:
AMHA here it makes more sense that the tests exercise the the hook under test with what will happen in real conditions.